src/DataReader/parse_queries.rs

```diff
@@ -10,18 +10,41 @@ pub struct QueryParser;
 ///This file handles parsing the queries based on the abstract syntax described in the .pest files in the grammar folder
 ///For clarification see documentation on pest crate
 
-pub fn parse(edge_attribute_str: &str) -> Result<QueryExpression, Error<Rule>> {
-    let mut pairs = QueryParser::parse(Rule::query, edge_attribute_str)
+pub fn parse(edge_attribute_str: &str) -> Vec<QueryExpression> {
+    let mut pairs = QueryParser::parse(Rule::queries, edge_attribute_str)
         .unwrap_or_else(|e| panic!("Could not parse as rule with error: {}", e));
     let pair = pairs.next().unwrap();
+    let mut queries = vec![];
     match pair.as_rule() {
-        Rule::query => Ok(build_query_from_pair(pair)),
+        Rule::queries => {
+            build_queries(pair, &mut queries);
+            queries
+        }
         err => {
             panic!("Unable to match query string as rule: {:?}", err)
         }
     }
 }
 
+pub fn build_queries(pair: pest::iterators::Pair<Rule>, list: &mut Vec<QueryExpression>) {
+    match pair.as_rule() {
+        Rule::queryList => {
+            for p in pair.into_inner() {
+                build_queries(p, list)
+            }
+        }
+        Rule::queries => {
+            for p in pair.into_inner() {
+                build_queries(p, list)
+            }
+        }
+        Rule::query => {
+            list.push(build_query_from_pair(pair));
+        }
+        other => {}
```

Comment:
Rust seems to like using `_ => {}` when matching all cases and the variable is unused.